compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -1071,12 +1071,17 @@ trait Checking {
   def checkValidInfix(tree: untpd.InfixOp, meth: Symbol)(using Context): Unit = {
     tree.op match {
       case id @ Ident(name: Name) =>
+        def methCompiledBeforeDeprecation =
+          meth.tastyVersion match
+            case Some(version) => version.minor < 4 // compiled before 3.4
```

Comment:
if we ever bump the major version you may forget to update this code

Reply:
I.e. in `29.3` this would return true, but really this method should become dead code

Reply:
I will add the major version to the condition